src/main/java/org/opensearch/flowframework/workflow/DeployModelStep.java

```diff
@@ -70,7 +68,7 @@ public void onFailure(Exception e) {
                 break;
             }
         }
-        machineLearningNodeClient.deploy(modelId, actionListener);
+        mlClient.deploy(modelId, actionListener);
```

Comment:
additionally here, we should complete exceptionally if the `model_id` field is not present